shared-components/statviz/components/visualizations/demographic/DemographicPyramid.tsx

```diff
@@ -45,28 +45,99 @@ export default function DemographicPyramid({
 }: IDemographicChartProps) {
   const onExport = getOnExport(BarChartCenterAxis);
 
-  const prepareFacts = () => {
+  const prepareFactsForGraph = () => {
     const dataXr = tidy(
       demographics?.facts as BeneficiaryDemographicsResult[],
-      filter((value) => value.gender === "Male"),
+      filter((value) => value.gender === "Male" && value.age !== null),
       groupBy("age", [summarize({ count: sum("count") })]),
       map((value) => ({ x: value.count, y: value.age ?? 0 })),
     );
 
     const dataXl = tidy(
       demographics?.facts as BeneficiaryDemographicsResult[],
-      filter((value) => value.gender === "Female"),
+      filter((value) => value.gender === "Female" && value.age !== null),
       groupBy("age", [summarize({ count: sum("count") })]),
       map((value) => ({ x: value.count, y: value.age ?? 0 })),
     );
 
     return [dataXr, dataXl];
   };
 
-  const [dataXr, dataXl] = useMemo(prepareFacts, [demographics?.facts]);
+  const prepareFactsForText = () => {
+    const totalCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    const maleCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      filter((value) => value.gender === "Male"),
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    const femaleCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      filter((value) => value.gender === "Male"),
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    const diverseCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      filter((value) => value.gender === "Diverse"),
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    const ageNullCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      filter((value) => value.age === null),
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    const ageNullOrDiverseCount = tidy(
+      demographics?.facts as BeneficiaryDemographicsResult[],
+      filter((value) => value.age === null || value.gender === "Diverse"),
+      summarize({ total: sum("count") }),
+    )[0].total;
+
+    return [totalCount, maleCount, femaleCount, diverseCount, ageNullCount, ageNullOrDiverseCount];
+  };
+
+  const [dataXr, dataXl] = useMemo(prepareFactsForGraph, [demographics?.facts]);
+
+  const [totalCount, maleCount, femaleCount, diverseCount, ageNullCount, ageNullOrDiverseCount] =
+    useMemo(prepareFactsForText, [demographics?.facts]);
+
+  const beneficiariesRegistrationsText = (
+    <Text>
+      There were <chakra.span as="b">{totalCount}</chakra.span> beneficiaries registered in the
+      selected time period, <chakra.span as="b">{maleCount}</chakra.span> were male and{" "}
+      <chakra.span as="b">{femaleCount}</chakra.span> were female.
+      {ageNullOrDiverseCount ? (
+        <chakra.span>
+          <br />
```

Comment:
Needs an additional `<br />` or if we have the the right CSS paragraph formatting, wrap the first line with `<p> </p>`, you tell me which it is